src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,158 @@
+    int offsetTotalMinutes = Integer.parseInt(timezoneOffset);
+    int offsetHours = offsetTotalMinutes / 60;
+    int offsetMinutes = offsetTotalMinutes % 60;
+    return ZoneOffset.ofHoursMinutes(offsetHours, offsetMinutes);
+  }
+
+  // Returns a readable date string such as "Tue 7/7".
+  private static String generateDate(String dayOfWeek, int month, int dayOfMonth) {
+    return dayOfWeek + " " + month + "/" + dayOfMonth;
+  }
+
+  // This method returns a list of UTC Strings for a single day, creating one String per Time
+  // Slot.
+  private static List<String> generateUTCEncodings(
+      int year, int month, int dayOfMonth, ZoneId zoneId) {
+    List<String> utcEncodings = new ArrayList<String>();
+    int[] minutes = {0, 15, 30, 45};
+    for (int i = TIME_8AM; i <= TIME_7PM; i++) {
+      for (int j = 0; j < 4; j++) {
+        LocalDateTime localTimeSlot = LocalDateTime.of(year, month, dayOfMonth, i, minutes[j]);
+        ZonedDateTime utcTimeSlot =
+            ZonedDateTime.of(localTimeSlot, zoneId).withZoneSameInstant(ZoneOffset.UTC);
+        String formattedUTCTimeSlot = utcTimeSlot.format(DateTimeFormatter.ISO_INSTANT);
+        utcEncodings.add(formattedUTCTimeSlot);
+      }
+    }
+    return utcEncodings;
+  }
+
+  // TODO: Access the time slots from data store to tell if they are selected or not.
+  // This method will tell whether or not a time slot has already been selected. (See
+  // TODO above).
+  private static List<Boolean> getSelectedStatuses() {
+    List<Boolean> selectedStatuses = new ArrayList<Boolean>();
+    for (int i = 0; i < numberOfSlotsPerDay; i++) {
+      selectedStatuses.add(false);
+    }
+    return selectedStatuses;
+  }
+
+  // This method combines the lists of previously generated information into a list of constructed
+  // AvailabilityTimeSlot objects.
+  private static List<AvailabilityTimeSlot> generateTimeSlots(
+      List<String> utcEncodings, List<String> times, String date, List<Boolean> selectedStatuses) {
+    List<AvailabilityTimeSlot> availabilityTimeSlots = new ArrayList<AvailabilityTimeSlot>();
+    for (int i = 0; i < numberOfSlotsPerDay; i++) {
+      availabilityTimeSlots.add(
+          AvailabilityTimeSlot.create(
+              utcEncodings.get(i), times.get(i), date, selectedStatuses.get(i)));
+    }
+    return availabilityTimeSlots;
+  }
+
+  // TODO: Change to generate current week with calls to generateDate
+  private static List<String> generateDates() {
+    List<String> dates = new ArrayList<String>();
+    dates.add("Sunday 6/28");
+    dates.add("Monday 6/29");
+    dates.add("Tuesday 6/30");
+    dates.add("Wednesday 7/1");
+    dates.add("Thursday 7/2");
+    dates.add("Friday 7/3");
+    dates.add("Saturday 7/4");
+    return dates;
+  }
+
+  // Returns a list of readable time Strings such as "8:00 AM".
+  private static List<String> generateTimes() {
+    List<String> times = new ArrayList<String>();
+    for (int i = 8; i < 12; i++) {
```

Comment:
This is probably supposed to be using your constants.

Reply:
Becomes moot with new implementation of availableStartTimes().